cmd/rigs/cmd/publish_data.go

```diff
@@ -56,35 +61,37 @@ var dataCmd = &cobra.Command{
 			jobs = append(jobs, layersJobs...)
 		}
 
+		if viper.GetBool("rigs") || publishAll {
+			rigsJobs, err := rigsJobs(ctx, localStore, &jobID)
+			checkErr(err)
+			jobs = append(jobs, rigsJobs...)
+		}
+
 		if viper.GetBool("attrs") || publishAll {
 			attsJobs, err := attrsJobs(ctx, localStore, &jobID)
 			checkErr(err)
 			jobs = append(jobs, attsJobs...)
 		}
 
+		if viper.GetBool("deals") || publishAll {
+			dealsJobs, err := dealsJobs(ctx, localStore, &jobID)
+			checkErr(err)
+			jobs = append(jobs, dealsJobs...)
+		}
+
 		if viper.GetBool("lookups") || publishAll {
 			jobs = append(jobs, lookupsJob(localStore, &jobID))
 		}
 
 		pool := wpool.New(viper.GetInt("concurrency"), rate.Every(time.Millisecond*200))
 		go pool.GenerateFrom(jobs)
 		go pool.Run(ctx)
-	Loop:
-		for {
-			select {
-			case r, ok := <-pool.Results():
-				if !ok {
-					continue
-				}
-				if r.Err != nil {
-					fmt.Printf("error processing job %d: %v\n", r.ID, r.Err)
-					continue
-				}
-				fmt.Printf("processed job %d. %s\n", r.ID, r.Desc)
-			case <-pool.Done:
-				fmt.Println("done")
-				break Loop
+		for r := range pool.Results() {
+			if r.Err != nil {
+				fmt.Printf("error processing job %d: %v\n", r.ID, r.Err)
+				continue
 			}
+			fmt.Printf("processed job %d. %s\n", r.ID, r.Desc)
```

Comment:
Cleaned up the API for my worker pool utility so this loop is no longer necessary. So you'll see diffs like this all over the PR, don't worry about them.